sale_timesheet_project_link/models/sale_order.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2018 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+
+
+class SaleOrder(models.Model):
+
+    _inherit = 'sale.order'
+
+    all_project_ids = fields.Many2many(
+        comodel_name='project.project',
+        compute='_compute_all_projects',
+    )
+    all_projects_count = fields.Integer(
+        compute='_compute_projects_count',
+    )
+
+    @api.multi
+    @api.depends(
+        'order_line.task_id.project_id',
+        'project_ids',
+    )
+    def _compute_all_projects(self):
+        for rec in self:
+            rec.all_project_ids = (
+                rec.order_line.mapped('task_id.project_id') | rec.project_ids)
```

Comment:
Is a computed field the best way to do this? A simple method would be less stressful for the orm cache.

Reply:
Yes I don't really need this field, it is only used when counting the number of projects to show the button in the form view. I will change it for a simple method.